client/my-sites/comments/comment/comment-html-editor/index.jsx

```diff
@@ -0,0 +1,242 @@
+			paragraph: false,
+			selfClosed: false,
+			text: null,
+		}
+	) => {
+		const element = document.createElement( tag );
+		each( attributes, ( value, key ) => element.setAttribute( key, value ) );
+		element.innerHTML = '<!---->';
+		const fragments = element.outerHTML.split( '<!---->' );
+		const opener =
+			( options.paragraph ? '\n' : '' ) +
+			( options.indent ? '\t' : '' ) +
+			fragments[ 0 ] +
+			( options.paragraph ? '\n' : '' );
+		const closer =
+			fragments[ 1 ] + ( options.newLineAfter ? '\n' : '' ) + ( options.paragraph ? '\n\n' : '' );
+		const inner = options.text || this.splitSelectedContent().inner;
+
+		if ( inner.length || options.alsoClose ) {
+			return this.insertContent( opener + inner + closer, options.adjustCursorPosition );
+		}
+
+		if ( ! options.selfClosed && this.isTagOpen( tag ) ) {
+			this.setState( ( { openTags } ) => ( { openTags: filter( openTags, tag ) } ) );
+			return this.insertContent( closer, options.adjustCursorPosition );
+		}
+
+		if ( ! options.selfClosed ) {
+			this.setState( ( { openTags } ) => ( { openTags: openTags.concat( tag ) } ) );
+		}
+		return this.insertContent( opener, options.adjustCursorPosition );
+	};
+
+	insertContent = ( content, adjustCursorPosition = 0 ) => {
+		const userAgent = get( window, 'navigator.userAgent', '' );
+
+		// In Firefox, IE, and Edge, `document.execCommand( 'insertText' )` doesn't work.
+		// @see https://bugzilla.mozilla.org/show_bug.cgi?id=1220696
+		if (
+			/(?:firefox|fxios)/i.test( userAgent ) ||
+			/(?:edge|msie |trident.+?; rv:)/i.test( userAgent )
+		) {
+			const { selectionEnd, value } = this.textarea;
+			const { before, after } = this.splitSelectedContent();
+			const newContent = before + content + after;
+			const event = { target: { value: newContent } };
+			this.props.onChange( event, () => {
+				this.setCursorPosition(
+					selectionEnd,
+					newContent.length - value.length + adjustCursorPosition
+				);
+				this.textarea.focus();
+			} );
+			return;
+		}
+
+		this.textarea.focus();
+		document.execCommand( 'insertText', false, content );
+
+		if ( adjustCursorPosition ) {
+			this.setCursorPosition( this.textarea.selectionEnd, adjustCursorPosition );
+		}
+	};
+
+	insertStrongTag = () => this.insertHtmlTag( 'strong' );
+
+	insertEmTag = () => this.insertHtmlTag( 'em' );
+
+	insertATag = ( attributes, text ) => {
+		if ( text ) {
+			return this.insertHtmlTag( 'a', attributes, { text } );
+		}
+		// Also move the cursor inside <a></a>
+		this.insertHtmlTag( 'a', attributes, { adjustCursorPosition: -4, alsoClose: true } );
+	};
+
+	insertBlockquoteTag = () => this.insertHtmlTag( 'blockquote', {}, { paragraph: true } );
+
+	insertDelTag = () => this.insertHtmlTag( 'del', { datetime: this.props.moment().format() } );
+
+	insertInsTag = () => this.insertHtmlTag( 'ins', { datetime: this.props.moment().format() } );
+
+	insertImgTag = attributes => this.insertHtmlTag( 'img', attributes, { selfClosed: true } );
```

Comment:
note that as an empty element this needs no `selfClosed` attribute - we should know that inside of `insertHtmlTag` because `fragments[1]` will be undefined.